### PR TITLE
[DORT] Use new FX-to-ONNX exporter

### PR DESCRIPTION
**Pull request by wschin**
The ONNX exporter in DORT have been moved to PyTorch as a formal feature. We therefore switch to consume the exporter from PyTorch instead of maintaining two duplicates.

### CODE REVIEW

File: orttraining/orttraining/python/training/torchdynamo/ort_backend.py [hunk 1]
```diff
@@ -16,18 +16,47 @@
...
 import onnxruntime  # type: ignore
 from onnxruntime.capi import _pybind_state as ORTC
 
+# DEFAULT_ONNX_EXPORTER_OPTIONS contains shared information between exporter and DORT.
+# For example, they should use the same decomposition table to maintain the same set
+# operators when
+#  1. capturing FX graph in torch.compile
+#  2. call exporter's API to convert `torch.fx.GraphModule` to ONNX model.
+DEFAULT_ONNX_EXPORTER_OPTIONS = torch.onnx._internal.exporter.ResolvedExportOptions(
+    torch.onnx._internal.exporter.ExportOptions()
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/torchdynamo/ort_backend.py:42]: `ResolvedExportOptions` was meant to be the private (and consolidated version) of `ExportOptions`. Why not use `ExportOptions` directly?

We usually go with `ResolvedExportOptions` when we want to change a private attribute for debugging or accessing an invisible feature.

Maybe `onnx_registry` and `decomposition_table` should be public properties to prevent the `ResolvedExportOptions(ExportOptions())` pattern to hack the internals?

File: orttraining/orttraining/python/training/torchdynamo/ort_backend.py [hunk 2]
```diff
@@ -536,17 +395,19 @@ def _ort_acclerated_call(self, graph_module: torch.fx.GraphModule, *args, **kwar
...
-            else:
-                _decorate_script_module(script_module, args, (prim_outputs,))
-            # Generate ONNX ModelProto from torch._C.Graph.
-            onnx_proto = _create_onnx_proto(script_module)
+
+            diagnostic_context = torch.onnx._internal.diagnostics.infra.DiagnosticContext(
+                name="OrtBackend",
+                version="0.0.1",
+            )
+            exported = torch.onnx._internal.fx.passes.export_fx_to_onnxscript(
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/torchdynamo/ort_backend.py:403]: `export_fx_to_onnxscript` no longer exists

File: orttraining/orttraining/python/training/torchdynamo/ort_backend.py (hunk 1 above)
Comment (wschin) [reply orttraining/orttraining/python/training/torchdynamo/ort_backend.py:42]: Well, DORT is a kind of internal component of FX-to-ONNX exporter. I personally won't bother for this. :)

File: orttraining/orttraining/python/training/torchdynamo/ort_backend.py (hunk 2 above)
Comment (wschin) [reply orttraining/orttraining/python/training/torchdynamo/ort_backend.py:403]: I feel sad now. lol.

File: orttraining/orttraining/python/training/torchdynamo/ort_backend.py [hunk 3]
```diff
@@ -16,18 +16,47 @@
 import torch.fx
 import torch.jit
 import torch.onnx
+import torch.onnx._internal
```
[the remark this answers is not in the document]
Comment (wschin) [reply orttraining/orttraining/python/training/torchdynamo/ort_backend.py:19]: Maybe not. There are different kinds of `_internal` spreading in the code. If we label it this way, we need to label A LOT of lines with this comment. It will look funny.

[the remark this answers is not in the document]
Comment (wschin) [reply orttraining/orttraining/python/training/torchdynamo/ort_backend.py:19]: The FIX will automatically happen since otherwise the code throws.

[the remark this answers is not in the document]
Comment (justinchuby) [reply orttraining/orttraining/python/training/torchdynamo/ort_backend.py:19]: I just meant this blob since they seem to be clustered together here in the import section. We also don't want readers to stumble on this code and think importing from _internal is the right way to develop with torch.onnx. I was going to propose all caps "THIS IS DANGEROUS" sort of thing in the comment but at least use https://github.com/microsoft/onnxruntime/issues/11901 to track?

File: tools/ci_build/github/azure-pipelines/linux-cpu-eager-pipeline.yml [hunk 4]
```diff
@@ -69,7 +69,8 @@ jobs:
             bash -c "
               export PYTHONPATH=/build/Release && \
               /opt/python/cp39-cp39/bin/python3.9 -m pip install /build/Release/dist/*.whl && \
-              /opt/python/cp39-cp39/bin/python3.9 /onnxruntime_src/orttraining/orttraining/test/python/orttraining_test_dort.py"
+              /opt/python/cp39-cp39/bin/python3.9 /onnxruntime_src/orttraining/orttraining/test/python/orttraining_test_dort.py && \"
+              /opt/python/cp39-cp39/bin/python3.9 /onnxruntime_src/orttraining/orttraining/test/python/orttraining_test_dort_custom_op.py"
```
Comment (wschin) [inline tools/ci_build/github/azure-pipelines/linux-cpu-eager-pipeline.yml:73]: @onnxruntime-es, this is the only es-relevant change.

**Review by jchen351 — APPROVED** [review]
